hudi-utilities/src/main/java/org/apache/hudi/utilities/transform/ChainedTransformer.java

```diff
@@ -93,9 +103,13 @@ public List<String> getTransformersNames() {
   @Override
   public Dataset<Row> apply(JavaSparkContext jsc, SparkSession sparkSession, Dataset<Row> rowDataset, TypedProperties properties) {
     Dataset<Row> dataset = rowDataset;
+    Option<Schema> incomingSchemaOpt = sourceSchemaOpt;
     for (TransformerInfo transformerInfo : transformers) {
       Transformer transformer = transformerInfo.getTransformer();
       dataset = transformer.apply(jsc, sparkSession, dataset, transformerInfo.getProperties(properties));
```

Comment:
just a reminder from our discussion, this will likely through a logical plan error and not enter the section below

Reply:
reminder that we don't need to validate the schema but rather validate that the transformers will result in a valid plan. if they do not, we want to throw an easy to understand/debug error with context for a developer to understand why it's not working.

Reply:
I was able to find a way to print the missing columns, new columns and input columns within the exception message. Please check `org.apache.hudi.utilities.transform.ChainedTransformer#printLogicalPlanErrorDetails`. If it looks good I can also include it in the exception metadata.

Reply:
I have created new exception class `HoodieTransformerSchemaException` and added it as metadata there.